tests/Unit/Cache/ElementCacheTagCheckerTest.php

```diff
@@ -0,0 +1,397 @@
+<?php declare(strict_types=1);
+
+namespace Neusta\Pimcore\HttpCacheBundle\Tests\Unit\Cache;
+
+use Neusta\Pimcore\HttpCacheBundle\Cache\CacheTag;
+use Neusta\Pimcore\HttpCacheBundle\Cache\CacheTagChecker;
+use Neusta\Pimcore\HttpCacheBundle\Cache\CacheTagChecker\ElementCacheTagChecker;
+use Neusta\Pimcore\HttpCacheBundle\Cache\CacheType\CustomCacheType;
+use Neusta\Pimcore\HttpCacheBundle\Element\ElementRepository;
+use PHPUnit\Framework\TestCase;
+use Pimcore\Model\Asset;
+use Pimcore\Model\DataObject;
+use Pimcore\Model\Document;
+use Prophecy\PhpUnit\ProphecyTrait;
+use Prophecy\Prophecy\ObjectProphecy;
+
+final class ElementCacheTagCheckerTest extends TestCase
+{
+    use ProphecyTrait;
+
+    /** @var ObjectProphecy<CacheTagChecker> */
+    private ObjectProphecy $innerCacheTagChecker;
+
+    /** @var ObjectProphecy<ElementRepository> */
+    private ObjectProphecy $elementRepository;
+
+    protected function setUp(): void
+    {
+        $this->innerCacheTagChecker = $this->prophesize(CacheTagChecker::class);
+        $this->elementRepository = $this->prophesize(ElementRepository::class);
+    }
+
+    /**
+     * @test
+     */
+    public function it_delegates_cache_tag_check_to_next_cache_tag_checker(): void
+    {
+        $tag = CacheTag::fromString('foo', new CustomCacheType('custom'));
+        $elementCacheTagChecker = new ElementCacheTagChecker(
+            $this->innerCacheTagChecker->reveal(),
+            $this->elementRepository->reveal(),
+            ['enabled' => true, 'types' => ['foo' => true]],
+            ['enabled' => false],
+            ['enabled' => false],
```

Comment:
What about using named arguments here so we can better see which elements are enabled or disabled?